test/integration/client/exec_test.go

```diff
@@ -0,0 +1,329 @@
+								"clientKeyData": %q
+							}
+						}`, clientAuthorizedToken, string(unauthorizedCert), string(unauthorizedKey)),
+				})
+			},
+		},
+		{
+			name: "unauthorized token and unauthorized certificate",
+			clientConfigFunc: func(c *rest.Config) {
+				c.ExecProvider.Env = append(c.ExecProvider.Env, clientcmdapi.ExecEnvVar{
+					Name: outputEnvVar,
+					Value: fmt.Sprintf(`{
+							"kind": "ExecCredential",
+							"apiVersion": "client.authentication.k8s.io/v1beta1",
+							"status": {
+								"token": "%s",
+								"clientCertificateData": %q,
+								"clientKeyData": %q
+							}
+						}`, clientUnauthorizedToken, string(unauthorizedCert), string(unauthorizedKey)),
+				})
+			},
+			wantClientErrorPrefix: "Unauthorized",
+		},
+		{
+			name: "good token with static auth basic creds favors static auth basic creds",
+			clientConfigFunc: func(c *rest.Config) {
+				c.ExecProvider.Env = append(c.ExecProvider.Env, clientcmdapi.ExecEnvVar{
+					Name: outputEnvVar,
+					Value: fmt.Sprintf(`{
+							"kind": "ExecCredential",
+							"apiVersion": "client.authentication.k8s.io/v1beta1",
+							"status": {
+								"token": "%s"
+							}
+						}`, clientAuthorizedToken),
+				})
+				c.Username = "unauthorized"
+				c.Password = "unauthorized"
+			},
+			wantClientErrorPrefix: "Unauthorized",
+		},
+		{
+			name: "good token with static auth bearer token favors static auth bearer token",
+			clientConfigFunc: func(c *rest.Config) {
+				c.ExecProvider.Env = append(c.ExecProvider.Env, clientcmdapi.ExecEnvVar{
+					Name: outputEnvVar,
+					Value: fmt.Sprintf(`{
+							"kind": "ExecCredential",
+							"apiVersion": "client.authentication.k8s.io/v1beta1",
+							"status": {
+								"token": "%s"
+							}
+						}`, clientAuthorizedToken),
+				})
+				c.BearerToken = "some-unauthorized-token"
+			},
+			wantClientErrorPrefix: "Unauthorized",
+		},
+		{
+			name: "good token with static auth cert and key favors exec plugin",
+			clientConfigFunc: func(c *rest.Config) {
+				c.ExecProvider.Env = append(c.ExecProvider.Env, clientcmdapi.ExecEnvVar{
+					Name: outputEnvVar,
+					Value: fmt.Sprintf(`{
+							"kind": "ExecCredential",
+							"apiVersion": "client.authentication.k8s.io/v1beta1",
+							"status": {
+								"token": "%s"
+							}
+						}`, clientAuthorizedToken),
+				})
+				c.CertData = unauthorizedCert
+				c.KeyData = unauthorizedKey
+			},
+		},
+	}
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			tmpDir, err := ioutil.TempDir("", "kubernetes-client-exec-test-plugin-dir-*")
+			if err != nil {
+				t.Fatal(err)
+			}
+			defer os.RemoveAll(tmpDir)
+
+			clientConfig := rest.AnonymousClientConfig(result.ClientConfig)
+			clientConfig.ExecProvider = &clientcmdapi.ExecConfig{
+				Command: "testdata/exec-plugin.sh",
+				// TODO(ankeesler): move to v1 once exec plugins go GA.
+				APIVersion: "client.authentication.k8s.io/v1beta1",
+				Env:        []clientcmdapi.ExecEnvVar{},
```

Comment:
nit: no need to initialize the slice

Reply:
12dc50eff84ced8632cd0bd42a0f26a52007da45